Sources/BraveWallet/Crypto/Stores/AccountActivityStore.swift

```diff
@@ -67,101 +67,127 @@ class AccountActivityStore: ObservableObject {
   func update() {
     Task { @MainActor in
       let coin = account.coin
-      let network = await rpcService.network(coin)
-      let keyring = await keyringService.keyringInfo(coin.keyringId)
-      let allTokens: [BraveWallet.BlockchainToken] = await blockchainRegistry.allTokens(network.chainId, coin: network.coin)
-      let allUserAssets: [BraveWallet.BlockchainToken] = await walletService.userAssets(network.chainId, coin: network.coin)
-      let userVisibleTokens = allUserAssets.filter(\.visible)
-      self.assets = await fetchAssets(userVisibleTokens: userVisibleTokens, network: network)
-      self.NFTAssets = await fetchNFTAssets(userVisibleTokens: userVisibleTokens, network: network)
-      let assetRatios = self.assets.reduce(into: [String: Double](), {
-        $0[$1.token.assetRatioId.lowercased()] = Double($1.price)
-      })
-      self.transactionSummaries = await fetchTransactionSummarys(
-        network: network,
-        accountInfos: keyring.accountInfos,
-        userVisibleTokens: allUserAssets,
-        allTokens: allTokens,
-        assetRatios: assetRatios
-      )
-    }
-  }
-
-  @MainActor private func fetchAssets(
-    userVisibleTokens: [BraveWallet.BlockchainToken],
-    network: BraveWallet.NetworkInfo
-  ) async -> [AssetViewModel] {
-    let visibleUserAssets = userVisibleTokens.filter { !$0.isErc721 && !$0.isNft }
-    var updatedAssets = visibleUserAssets.map {
-      AssetViewModel(token: $0, network: network, decimalBalance: 0, price: "", history: [])
-    }
-    // fetch price for each asset
-    let priceResult = await assetRatioService.priceWithIndividualRetry(
-      visibleUserAssets.map { $0.assetRatioId.lowercased() },
-      toAssets: [currencyFormatter.currencyCode],
-      timeframe: .oneDay
-    )
-    for price in priceResult.assetPrices {
-      if let index = updatedAssets.firstIndex(where: {
-        $0.token.assetRatioId.caseInsensitiveCompare(price.fromAsset) == .orderedSame
-      }) {
-        updatedAssets[index].price = price.price
+      let networks = await rpcService.allNetworks(coin)
+        .filter { $0.chainId != BraveWallet.LocalhostChainId } // localhost not supported
+      
+      struct NetworkAssets: Equatable {
+        let network: BraveWallet.NetworkInfo
+        let tokens: [BraveWallet.BlockchainToken]
+        let sortOrder: Int
       }
-    }
-    // fetch balance for each asset
-    typealias TokenBalance = (token: BraveWallet.BlockchainToken, balance: Double?)
-    let tokenBalances = await withTaskGroup(of: [TokenBalance].self) { @MainActor group -> [TokenBalance] in
-      for token in visibleUserAssets {
-        group.addTask { @MainActor in
-          let balance = await self.rpcService.balance(for: token, in: self.account, network: network)
-          return [TokenBalance(token, balance)]
+      let allVisibleUserAssets = await walletService.allVisibleUserAssets(in: networks)
+      let visibleUserAssetsForNetwork = allVisibleUserAssets.map {
```

Comment:
nit: How big is the `allVisibleUserAssets` here? I am asking because it might be better idea to run this loop once to get UserAssets and UserNFTAssets together in same loop instead of 2 (map)

Reply:
It starts relatively small, but can grow as user selects more assets. I think I can improve this to one 2-dimensional for loop and append to either `userVisibleAssets` / `userVisibleNFTs` depending on the token type. I think a couple loops here I can apply this to, thanks 👍

Reply:
Addressed in [b889280](https://github.com/brave/brave-ios/pull/6493/commits/b889280ca4ac41abb9927ae2c8012b68126b923c)